src/components/device/ha-device-automation-picker.ts

```diff
@@ -0,0 +1,173 @@
+import "@polymer/paper-input/paper-input";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-listbox/paper-listbox";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import {
+  DeviceAutomation,
+  deviceAutomationsEqual,
+} from "../../data/device_automation";
+import "../../components/ha-paper-dropdown-menu";
+
+const NO_AUTOMATION_KEY = "NO_AUTOMATION";
+const UNKNOWN_AUTOMATION_KEY = "UNKNOWN_AUTOMATION";
+
+export abstract class HaDeviceAutomationPicker<
+  T extends DeviceAutomation
+> extends LitElement {
+  public hass!: HomeAssistant;
+  @property() public label?: string;
+  @property() public deviceId?: string;
+  @property() public value?: T;
+  protected NO_AUTOMATION_TEXT = "No automations";
+  protected UNKNOWN_AUTOMATION_TEXT = "Unknown automation";
+  @property() private _automations: T[] = [];
+
+  // Trigger an empty render so we start with a clean DOM.
+  // paper-listbox does not like changing things around.
+  @property() private _renderEmpty = false;
+
+  private get _key() {
+    if (!this.value || deviceAutomationsEqual(this._noAutomation, this.value)) {
+      return NO_AUTOMATION_KEY;
+    }
+
+    const idx = this._automations.findIndex((automation) =>
+      // deviceAutomationsEqual/*<T>*/(automation, this.value!)
+      deviceAutomationsEqual(automation, this.value!)
+    );
+
+    if (idx === -1) {
+      return UNKNOWN_AUTOMATION_KEY;
+    }
+
+    return `${this._automations[idx].device_id}_${idx}`;
+  }
+
+  protected localizeDeviceAutomation(value) {
```

Comment:
I want to do `protected abstract localizeDeviceAutomation(value);`, but then the call to this.localizeDeviceAutomation() from render() does not work.